crates/oxc_isolated_declarations/src/types.rs

```diff
@@ -75,11 +75,13 @@ impl<'a> IsolatedDeclarations<'a> {
                 self.ast.atom(&literal.value.to_js_string()),
             ),
             // `["string"] -> string`
-            PropertyKey::StringLiteral(literal) => {
+            PropertyKey::StringLiteral(literal) if is_identifier_name(&literal.value) => {
                 self.ast.property_key_static_identifier(literal.span, literal.value.as_str())
```

Comment:
As you said in https://github.com/oxc-project/oxc/pull/11329#discussion_r2110723730, we should convert a pure number string into a `StringLiteral` rather than a `StaticIdentifier`, and please add a test for it.

Reply:
Oh, I see that a pure number string will fall back into `clone_in`, so it is still a `StringLiteral`.